### PR TITLE
Add convert method for convert expression

### LINKED ISSUE #13694: Support for convert expression in database
This is a (multiple allowed):

* [ ] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.x

### Enhancement description
Add `CONVERT` or `CAST` or both expression to use for
* Get date part from an datetime
* Convert int to varchar to use in concat
* etc...

### Usage:
Using `$query->newExpr()->convert('db_field', 'target_type')` to generate convert expression

### PR DESCRIPTION
**Pull request by kingofnull**
I just add the convert method as it requested in issue. It just make CAST(expression AS datatype) expression which should works on most DBMS' .

Resolves https://github.com/cakephp/cakephp/issues/13694

### CODE REVIEW

**Comment by othercorey** [comment]
This is missing test cases.

File: src/Database/Expression/QueryExpression.php [hunk 1]
```diff
@@ -532,6 +532,20 @@ public function equalFields($left, $right)
...
     }
 
+    /**
+     * Adds a new type convertion to the expression object in the form "CAST(field AS type)".
+     *
+     * @param string $field Database field to be cast against type
+     * @param string $type target data type to cast field to it
+     * @return $this
+     */
+    public function convert($field, $type)
```
Comment (ADmad) [inline src/Database/Expression/QueryExpression.php]: Why not name the function `cast()` to match the SQL function name?

Comment (kingofnull) [reply src/Database/Expression/QueryExpression.php]: the issue request convert name. because of portability I used the  cast instead of convert.

**Comment by dereuromark** [comment]
Should this go into 4.x first? And then get backported to 3.next (3.9)?

File: src/Database/Expression/QueryExpression.php [hunk 2]
```diff
@@ -532,6 +532,20 @@ public function equalFields($left, $right)
...
+    {
+        $this->add("CAST($field AS $type)");
```
Comment (ndm2) [inline src/Database/Expression/QueryExpression.php]: I have rather conflicted feelings about this. Building the SQL immediately basically turns the cast into a non-expression, as it cannot be traversed, transformed, etc., and it's different to what all other expression builder methods do. Furthermore this neither supports automatic identifier quoting, nor does it allow to pass in expressions for the field, which is different to other expressions too.

In my own projects I've used dedicated function expression classes for casts, as they can benefit from having return types applied.

Comment (ADmad) [reply src/Database/Expression/QueryExpression.php]: Yup, this looks pretty unsafe.

**Pull request closed without merging**